src/sentry/monitoring/queues.py

```diff
@@ -111,26 +111,31 @@ def get_queue_by_name(name):
 queue_monitoring_cluster = redis.redis_clusters.get(CLUSTER_NAME)
 
 
+def unhealthy_queue_key(queue_name: str) -> str:
+    return f"{KEY_NAME}:queue_name"
+
+
 def is_queue_healthy(queue_name: str) -> bool:
     # check if queue is healthy by pinging Redis
-    return not queue_monitoring_cluster.sismember(KEY_NAME, queue_name)
+    return not queue_monitoring_cluster.exists(unhealthy_queue_key(queue_name))
 
 
-def is_healthy(queue_size):
+def _is_healthy(queue_size) -> bool:
     return queue_size < UNHEALTHY_QUEUE_SIZE_THRESHOLD
 
 
-def update_queue_stats(redis_cluster, unhealthy) -> None:
+def _update_queue_stats(redis_cluster, unhealthy) -> None:
+    if not unhealthy:
+        return
+
     with redis_cluster.pipeline(transaction=True) as pipeline:
-        pipeline.delete(KEY_NAME)
-        if unhealthy:
-            pipeline.sadd(KEY_NAME, *unhealthy)
-            # expire in 1min if we haven't checked in
-            pipeline.expire(KEY_NAME, 60)
+        # can't use mset because it doesn't support expiry
+        for queue in unhealthy:
+            pipeline.set(unhealthy_queue_key(queue), "", ex=UNHEALTHY_QUEUE_CHECK_INTERVAL)
```

Comment:
hm, I would actually set a specific value, like `true` or something?

Reply:
In the low priority queue, we used "1" for this. I think I'll change it to that.